From 31c3a3b930cc493bae1bdc7505665b187a388038 Mon Sep 17 00:00:00 2001
From: Rasmus Thomsen <oss@cogitri.dev>
Date: Mon, 9 Mar 2020 19:23:36 +0100
Subject: [PATCH] Add support for building with meson

This makes it easier for distros to install this and provides a pkgconfig
file for other projects which want to use this together with
meson
---
 meson.build | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)
 create mode 100644 meson.build

diff --git a/meson.build b/meson.build
new file mode 100644
index 0000000..422c562
--- /dev/null
+++ b/meson.build
@@ -0,0 +1,17 @@
+project('openssld', ['d'], version: '1.1.0h')
+
+openssl_dep = dependency('openssl')
+
+install_subdir(
+	'deimos',
+	install_dir: join_paths(get_option('includedir'), 'd', 'openssld'),
+)
+
+pkgc = import ('pkgconfig')
+
+pkgc.generate(
+	name: 'openssld',
+	subdirs: 'd/openssld',
+	requires: openssl_dep,
+	description: 'OpenSSL D interface',
+)
